CK3toEU4/Source/CK3World/Titles/Title.cpp

```diff
@@ -369,3 +368,53 @@ double CK3::Title::getBuildingWeight(const mappers::DevWeightsMapper& devWeights
 								 devWeightsMapper.getDevFromDev() * development;
 	return totalDev;
 }
+
+void CK3::Title::relinkDeFactoVassals()
+{
+	// We're reconstructing defacto hierarchy; we're redirecting our DFvassals from our DFtitle into our DJtitle - if holder of this title holds hold both.
+	// This is important - otherwise secondary titles wouldn't have their own vassals under them but under primary title, breaking PU splitoffs,
+	// and resulting with those secondary duchies/kingdoms being discarded as landless.
+
+	if (name.starts_with("c_") || name.starts_with("b_"))
+		return; // don't bother with counties and below.
+
+	for (const auto& dfVassal: dfVassals)
+	{
+		if (!dfVassal.second)
+			continue;
+
+		if (dfVassal.second->getDFLiege() && dfVassal.second->getDJLiege() && dfVassal.second->getDFLiege()->first != dfVassal.second->getDJLiege()->first)
+		{
+			// Do we own both titles?
+			if (holder && holder->second && holder->second->getDomain())
+			{
+				const auto holderTitles = holder->second->getDomain()->getDomain();
```

Comment:
The double getDomain() has left me lost. It might be better to use more variables just so maintainers can follow the flow better.

Reply:
Yes, this is how the game uses its objects and stores them in the save. domain is the data block under character that holds capital, laws in his domain, government form, but it also holds actual domain which is a list of llongs that refer to title IDs he owns.

I built the converter fast and dirty so its internal structure matches savegame structure, and didn't have time to go back and consolidate all those objects, which is what I will be doing once I get back to the converter.

For now the best I can do is rename the functions so they're differently named.